exceptions.py

```diff
@@ -243,12 +249,12 @@ def _let_one_through(exception, data):
     if my_sem.acquire(blocking=False):
         # Look at me - I'm the captain now.
         return True
-        pass
     else:
         # Someone else is already handling
         # the exception. Sit here until they
         # release the semaphore.
         my_sem.acquire()
+        my_sem.release()
```

Comment:
The previous logic was a bit convoluted. There was race condition happening which could cause multiple greenlets to succeed on `let_one_through`